### PR TITLE
docs: clarify behavior of ng serve with server.ts

### PR DESCRIPTION
**Pull request by hybrist**
See: https://stackoverflow.com/questions/77468756/how-to-use-express-server-in-dev-mode-in-angular-17-application-builder
See: https://github.com/angular/angular/issues/52929#issuecomment-1818078952

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jkrems** [comment]
Checked with @alan-agius4 to confirm that this is indeed expected behavior (for now) and the docs are outdated.

File: adev/src/content/guide/ssr.md [hunk 1]
```diff
@@ -45,6 +45,8 @@ To verify that the application is server-side rendered, run it locally with `ng
 
 ## Configure server-side rendering
 
+Note: In Angular v17 and later, `server.ts` is no longer used by `ng serve`. The dev server will use `main.server.ts` directly to set up `CommonEngine`.
```
Comment (alan-agius4) [inline adev/src/content/guide/ssr.md:48]: Remove `CommonEngine` as this is not used during `ng serve`.

```suggestion
Note: In Angular v17 and later, `server.ts` is no longer used by `ng serve`. The dev server will use `main.server.ts` directly to perfom server side rendering.
```

**Review by alan-agius4 — APPROVED** [review]
NIT, Otherwise LGTM

**Review by alan-agius4 — APPROVED** [review]

**Comment by alan-agius4** [comment]
**Cakretaker note:** my approval should suffice for this change.

**Review by thePunderWoman — APPROVED** [review]

**Comment by thePunderWoman** [comment]
This PR was merged into the repository by commit c3e5b0a0b0a0a0a6abf86dd56ab27c54777c8af2.

The changes were merged into the following branches: main, 18.2.x

**Pull request closed without merging**